js/main.js

```diff
@@ -115,14 +115,34 @@
             $("html, body").scrollTop(0);
 
             OCA.Onlyoffice.folderUrl = location.href;
-            window.history.pushState(null, null, url);
+
+
+            var wrapper = $('<div class="onlyoffice-header" />')
+            var btnClose = $('<a class="icon icon-close-white"></a>');
+            btnClose.on('click', function() {
+                OCA.Onlyoffice.CloseEditor();
+            });
+            if (OCA.Files.Sidebar) {
+                var btnShare = $('<a class="icon icon-menu-sidebar"></a>');
+                btnShare.on('click', function () {
+                    OCA.Files.Sidebar.file === "" ? OCA.Files.Sidebar.open(fileDir + '/' + fileName) : OCA.Files.Sidebar.close()
+                })
+                wrapper.prepend(btnShare)
+            }
+            wrapper.prepend(btnClose)
+            $('.header-right').append(wrapper)
+
+            if (typeof OCA.Files.Sidebar === 'undefined') {
```

Comment:
I change the link regardless of the existence of the sidebar

Reply:
The issue with keeping the link change is, that if the user reloads the page, the files app code whill not be loaded and therefore the sidebar will not work.

Reply:
The page with the editor does not allow to open the sidebar.
But having a link in the address bar is very important for reopening and copying the link

Reply:
Ok, i have an idea how to solve this then. I'll give that a try.